llm/cpp/llm.cpp

```diff
@@ -50,61 +33,50 @@ int main(int argc, char* argv[]) try {
     ov::InferRequest detokenizer = core.compile_model(argv[3], "CPU").create_infer_request();
     std::shared_ptr<ov::Model> model = core.read_model(argv[1]);
     constexpr size_t BATCH_SIZE = 1;
-    std::map<std::string, ov::PartialShape> shapes = {
-        {"input_ids", ov::PartialShape{
-            BATCH_SIZE, {1, std::numeric_limits<ov::Dimension::value_type>::max()}
+    std::map<size_t, ov::PartialShape> shapes = {
+        {0, ov::PartialShape{
+            BATCH_SIZE, -1
+        }},
+        {1, ov::PartialShape{
+            BATCH_SIZE, -1
         }},
-        {"attention_mask", ov::PartialShape{
-            BATCH_SIZE, {1, std::numeric_limits<ov::Dimension::value_type>::max()}
+        {2, ov::PartialShape{
+	    BATCH_SIZE, -1
         }}
     };
-    for (const ov::Output<ov::Node>& input : model->inputs()) {
-        for (const std::string& name : input.get_names()) {
-            if (name.rfind("past_key_values", 0) == 0) {
-                ov::PartialShape shape = input.get_partial_shape();
-                shape[0] = BATCH_SIZE;
-                shapes.emplace(name, shape);
-                break;
-            }
-        }
+    std::vector<ov::Output<ov::Node>> inputs = model->inputs();
+    for (size_t idx = 3; idx < inputs.size(); ++idx) {
+        ov::PartialShape shape = inputs.at(idx).get_partial_shape();
+        shape[0] = BATCH_SIZE;
+        shapes.emplace(idx, shape);
     }
     model->reshape(shapes);
-    ov::preprocess::PrePostProcessor p3(model);
-    p3.input("input_ids").tensor().set_element_type(ov::element::i32);  // cast to the type of tokenyzer's output
-    p3.input("attention_mask").tensor().set_element_type(ov::element::i32);
-    model = p3.build();
-    ov::InferRequest ireq = core.compile_model(model, "CPU", {ov::cache_dir("llm-cache")}).create_infer_request();
-    for (const ov::Output<ov::Node>& input : model->inputs()) {
-        for (const std::string& name : input.get_names()) {
-            if (name.rfind("past_key_values", 0) == 0) {
-                ireq.get_tensor(input).set_shape(input.get_partial_shape().get_min_shape());
-                break;
-            }
-        }
+    ov::InferRequest ireq = core.compile_model(model, "CPU", ov::cache_dir("llm-cache")).create_infer_request();
+    for (size_t idx = 2; idx < inputs.size(); ++idx) {
+        ireq.get_input_tensor(idx).set_shape(inputs.at(idx).get_partial_shape().get_min_shape());
     }
     ireq.get_tensor("input_ids").set_shape(input_ids.get_shape());  // TODO: replace with ireq.set_tensor("input_ids", input_ids); after it's fixed
     ireq.get_tensor("attention_mask").set_shape(input_ids.get_shape());
```

Comment:
should we use `attention_mask` instead of `input_ids` to configure `"attention_mask"` input?

Reply:
It doesn't matter, their shapes are identical. This code shouldn't have existed any way if there were no bugs in openvino